### PR TITLE
Showcases some of the new features from Spring Cloud Gateway for Kube…

### PR DESCRIPTION
**Pull request by jrshaffe**
Updates Animal Rescue to show off some of the SCG 2.1. features

- Animal Rescue Backend was updated to provide a `/api-docs` endpoint which is autogenerated using springdocs
- The `SpringCloudGatewayRouteConfig` was generated using the route conversion endpoint using the API docs mentioned above and documentation provided for it
- The `SpringCloudGatewayRouteConfig` for the animal rescue front end was updated to exclude openapi generation and mentioned in the readme
- The gateway-demo.yaml was updated to use the annotations for title, description, and documentation instead of the spec fields
- The OpenAPI generation filtering is mentioned in the readme with an example of how it can be used to get the OpenAPI docs specific for the Animal Rescue gateway demo.

To see what the changes to the `README.md` look like you can check out the branch https://github.com/spring-cloud-services-samples/animal-rescue/tree/jrshaffe/updateFor_v2.1.x

The generated API Doc for animal rescue can be seen here https://gist.github.com/jrshaffe/2fb2471e56dbf68b3d3bd1e2ddff2f26

### CODE REVIEW

File: backend/src/main/java/io/spring/cloud/samples/animalrescue/backend/domain/AnimalController.java [hunk 1]
```diff
@@ -64,6 +75,11 @@ public Mono<Void> submitAdoptionRequest(
 	}
 
 	@PutMapping("/animals/{animalId}/adoption-requests/{adoptionRequestId}")
+	@Operation(
+		summary = "Pet adoption API",
+		description = "Update pet adoption requests.",
+		tags = {"pet adoption"}
```
Comment (fombico) [inline backend/src/main/java/io/spring/cloud/samples/animalrescue/backend/domain/AnimalController.java:81]: I noticed the tags were missing in the generated json. Do you happen to know if this was due to `springdoc` or perhaps missing functionality on our side?

**Review by fombico — APPROVED** [review]

File: backend/src/main/java/io/spring/cloud/samples/animalrescue/backend/domain/AnimalController.java (hunk 1 above)
Comment (jrshaffe) [reply backend/src/main/java/io/spring/cloud/samples/animalrescue/backend/domain/AnimalController.java:81]: Good eye!  It looks like[ it's there in the docs generated in springdoc](https://gist.github.com/jrshaffe/22515c807fab0feb02b1687168fc2aa1#file-animal-rescue-from-springdoc-json-L26-L28).  I can take our generation logic.

Comment (jrshaffe) [reply backend/src/main/java/io/spring/cloud/samples/animalrescue/backend/domain/AnimalController.java:81]: I logged a [bug](https://app.zenhub.com/workspaces/tanzu---api-lifecycle-5c929bca8dcdbd6b61379e7c/issues/gh/pivotal-cf/spring-cloud-gateway-k8s/3898) for our openapi route conversion

**Review by csterwa — APPROVED** [review]
@jrshaffe was there anything that you wish was simpler in the UX for using OpenAPI specifications to generate API routes?

**Comment by jrshaffe** [comment]
> @jrshaffe was there anything that you wish was simpler in the UX for using OpenAPI specifications to generate API routes?

@csterwa One thing that might make it easier would be if there was some way infer some of the security settings for endpoints based off the openAPI spec.  For instance if I enable SSO at the service level, if there was a way to tell from the OpenAPI doc which endpoints didn't need SSO, if we could generate an exclusion for them automatically, or something similar with Token Relay

**Pull request merged**